src/torrent.cpp

```diff
@@ -2833,6 +2839,15 @@ bool is_downloading_state(int const st)
 				valid_endpoints++;
 			});
 
+#if TORRENT_USE_RTC
+			std::string const ae_protocol = ae.url.substr(0, ae.url.find(':'));
```

Comment:
I think it may be better to use `parse_url_components()` here. But, isn't there a place where tracker URLs are parsed and switched to use the corresponding `tracker_connection` object? would this not fit into that model?

Reply:
There is, in `tracker_manager.cpp`, however it's done too late. The issue is, the classic behavior is to queue multiple tracker requests (one per listen endpoint), however WebTorrent trackers do not care about it, so only one request should be done.

Reply:
I guess the `m_rtc_signalling` isn't a normal tracker, since it's a persistent connection

Reply:
Indeed, the WebSocket tracker serves for normal request and responses, but also transmits WebRTC offers and answers in both directions. That's why it needs to stay connected.